compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -291,18 +320,23 @@ impl<'a> ModCollector<'a> {
         &mut self,
         context: &mut Context,
         functions: Vec<NoirFunction>,
+        krate: CrateId,
         errors: &mut Vec<FileDiagnostic>,
     ) {
         let mut unresolved_functions =
             UnresolvedFunctions { file_id: self.file_id, functions: Vec::new() };
 
+        let module = ModuleId { krate, local_id: self.module_id };
+
         for function in functions {
             let name = function.name_ident().clone();
+            let func_id = context.def_interner.push_empty_fn();
+            let is_public = function.def.is_public;
+            let name_string = name.0.contents.clone();
 
             // First create dummy function in the DefInterner
             // So that we can get a FuncId
-            let func_id = context.def_interner.push_empty_fn();
-            context.def_interner.push_function_definition(name.0.contents.clone(), func_id);
+            context.def_interner.push_function_definition(name_string, func_id, is_public, module);
```

Comment:
I think we should handle all the attributes similarly, instead of handling 'pub' as a special case.
I.e we would assign them here as well.

Reply:
Ok, I moved several other attributes into the interner as well